proxy/proxy.go

```diff
@@ -563,16 +695,27 @@ func (c *client) interceptSystemQuery(hdr *frame.Header, stmt interface{}) {
 				c.send(hdr, &message.Invalid{ErrorMessage: err.Error()})
 			} else {
 				var data []message.Row
-				if parser.IsCountStarQuery(s) { // COUNT(*) always returns a value, even when there are no rows
-					data = []message.Row{{encodedZeroValue}}
+				for _, n := range c.proxy.nodes {
+					if n != c.proxy.localNode {
+						var row message.Row
+						row, err = c.filterSystemPeerValues(s, n, len(c.proxy.nodes)-1)
+						if err != nil {
+							break
+						}
+						data = append(data, row)
+					}
+				}
+				if err != nil {
+					c.send(hdr, &message.Invalid{ErrorMessage: err.Error()})
+				} else {
+					c.send(hdr, &message.RowsResult{
+						Metadata: &message.RowsMetadata{
+							ColumnCount: int32(len(columns)),
+							Columns:     columns,
+						},
+						Data: data,
+					})
```

Comment:
Does this take into account queries to `system.peers` with a `WHERE` clause? I think the java driver does it when reacting to topology protocol events but I'm not 100% sure. 

In ZDM we don't take this into account either and we haven't encountered an issue yet so this is just something to keep in mind if it does pop up eventually.

Reply:
No, it doesn't. It's just ignored, but I've verified it causes no issues.